### PR TITLE
Make sure the default os is in the list of available oses.

### PR DESCRIPTION
**Pull request by galthaus**
If not set it to the first in the list.

### CODE REVIEW

File: rails/app/models/barclamp_provisioner/base_images.rb [hunk 1]
```diff
@@ -25,4 +25,14 @@ def sysdata(nr)
     }
   end
 
+  def on_active(nr)
+    # Make sure that the default is in the list, if not reset it to the first entry.
+    oses = Attrib.get("provisioner-available-oses", admin).map{ |k,v| k } rescue []
+    default = Attrib.get('provisioner-target_os', Role.find_by(name: 'provisioner-os-install')) rescue "fred"
```
Comment (VictorLowther) [inline rails/app/models/barclamp_provisioner/base_images.rb]: fred? Why fred?

Comment (galthaus) [inline rails/app/models/barclamp_provisioner/base_images.rb]: what the code had (in the UI space) and will likely always fail the include test

**Comment by zehicle** [comment]
Likely there is a bug to make this dynamic

**Comment by zehicle** [comment]
Hi,

I'd like to add you to my professional network on LinkedIn.

- Rob

Confirm that you know Rob: https://www.linkedin.com/e/v2?e=0-im6cza5d-2g&t=ssuw&tracking=eml-guest-invite-cta&ek=invite_guest&sharedKey=5Ee2B_CK&invitationId=6118771196352880641








You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=0-im6cza5d-2g&t=lun&midToken=AQFmrFVhGEd36Q&ek=invite_guest&loid=AQHqBmkOUBNfNwAAAVOo-G9C1cCB3rHe7o__pTer5why9SBqjU6_e5W9g45J222Ky9yWLsjKpAXFGahpbna-GLqbLB-Uesph3D3JA5sN0lEevnXCR6WVo7FEMz_wR7ZA4qyysQu8oxzAb5BB-mBNZU5MWUG3znBqlvsDMCWqsDhTA_OSZw-OzU9PDPxIdzzLcqL-&eid=0-im6cza5d-2g


This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=0-im6cza5d-2g&a=customerServiceUrl&ek=invite_guest

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Comment by zehicle** [comment]
Rob Hirschfeld would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/e/v2?e=-rvrb6o-imfatvum-qq&a=preRegInvite&ek=first_guest_reminder_01&li=14&m=hero&ts=accept_text&sharedKey=5Ee2B_CK&invitationID=6118771196352880641

View Rob Hirschfeld&#39;s profile: https://www.linkedin.com/e/v2?e=-rvrb6o-imfatvum-qq&a=preRegInvite&ek=first_guest_reminder_01&li=3&m=hero&ts=profile_text&sharedKey=5Ee2B_CK&invitationID=6118771196352880641





You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-rvrb6o-imfatvum-qq&t=lun&midToken=AQFmrFVhGEd36Q&ek=first_guest_reminder_01&li=16&m=unsub&ts=HTML&eid=-rvrb6o-imfatvum-qq&loid=AQHZKyRt4wB_NAAAAVPJ-RSG1hvvSUHSjmxSYq1ZNuIwXMh2sYK2qe1XVstsVvW3bfYZweSK184bj4cTYM01kWNsjSCzXq1NVCcIQCy86Ku0BF5eAmhnNiGknUV1mi-3vsHQaQ7m3HgA-xwHt7mQZVfjxJxp7h2WnFdI73xhI_3UVXiVdzjUIpDR_zyZM3fAj60m

This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-rvrb6o-imfatvum-qq&a=customerServiceUrl&ek=first_guest_reminder_01

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Comment by zehicle** [comment]
Rob Hirschfeld would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/e/v2?e=-rvrb6o-impb4u5p-o9&a=preRegInvite&ek=second_guest_reminder_01&li=13&m=hero&ts=accept_text&sharedKey=5Ee2B_CK&invitationID=6118771196352880641

View Rob Hirschfeld&#39;s profile: https://www.linkedin.com/e/v2?e=-rvrb6o-impb4u5p-o9&a=preRegInvite&ek=second_guest_reminder_01&li=3&m=hero&ts=profile_text&sharedKey=5Ee2B_CK&invitationID=6118771196352880641





You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-rvrb6o-impb4u5p-o9&t=lun&midToken=AQFmrFVhGEd36Q&ek=second_guest_reminder_01&li=15&m=unsub&ts=HTML&eid=-rvrb6o-impb4u5p-o9&loid=AQFTkzCNnp2qCAAAAVPuBMNzLESqjFgZ6tiapSXhTQ1-qTdHUZ6s-mq6qd9MOLI1gjZvYEdGV-8ROHFh5r4otZ-6L-rwp_yeJnP0AYN4vT6ZZyL6NdFD1sQ-suwnT-Q_ckbasX9fjlqaJ9kuRm7U2hIMdV6VKGt9Qz1ByQ9sS8qkbs2F4T6VydPNgjLXmW_V4988

This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-rvrb6o-impb4u5p-o9&a=customerServiceUrl&ek=second_guest_reminder_01

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Pull request merged**